ios/brave-ios/Sources/Brave/Frontend/Browser/SearchResultAdClickedInfoBar.swift

```diff
@@ -0,0 +1,171 @@
+
+    self.toastView.snp.makeConstraints { make in
+      make.left.right.height.equalTo(self)
+      self.animationConstraint =
+        make.top.equalTo(self).offset(SearchResultAdClickedInfoBarUX.toastHeight).constraint
+    }
+
+    self.snp.makeConstraints { make in
+      make.height.equalTo(SearchResultAdClickedInfoBarUX.toastHeight)
+    }
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  fileprivate func createView(
+    _ labelText: String
+  ) -> UIView {
+    let horizontalStackView = UIStackView()
+    horizontalStackView.axis = .horizontal
+    horizontalStackView.alignment = .center
+    horizontalStackView.spacing = SearchResultAdClickedInfoBarUX.toastPadding
+
+    let label = UITextView()
+    label.textAlignment = .left
+    label.textColor = .white
+    label.font = SearchResultAdClickedInfoBarUX.toastLabelFont
+    label.backgroundColor = SearchResultAdClickedInfoBarUX.toastBackgroundColor
+    label.isEditable = false
+    label.isScrollEnabled = false
+    label.isSelectable = true
+    label.delegate = self
+
+    let learnMoreText = Strings.learnMore.withNonBreakingSpace
+    let attributes: [NSAttributedString.Key: Any] = [
+      .foregroundColor: UIColor.white,
+      .font: SearchResultAdClickedInfoBarUX.toastLabelFont,
+    ]
+
+    let linkAttributes: [NSAttributedString.Key: Any] = [
+      .font: SearchResultAdClickedInfoBarUX.toastLabelFont,
+      .foregroundColor: UIColor.white,
+      .underlineStyle: 1,
+    ]
+    label.linkTextAttributes = linkAttributes
+
+    let nsLabelAttributedString = NSMutableAttributedString(
+      string: labelText,
+      attributes: attributes
+    )
+    let nsLinkAttributedString = NSMutableAttributedString(
+      string: learnMoreText,
+      attributes: linkAttributes
+    )
+
+    if let url = URL(string: SearchResultAdClickedInfoBarUX.learnMoreUrl) {
+      let learnMoreRange = NSRange(location: 0, length: learnMoreText.count)
+      nsLinkAttributedString.addAttribute(.link, value: url, range: learnMoreRange)
+      nsLabelAttributedString.append(nsLinkAttributedString)
+      label.isUserInteractionEnabled = true
+    }
+    label.attributedText = nsLabelAttributedString
+
+    horizontalStackView.addArrangedSubview(label)
+
+    if let buttonImage = UIImage(braveSystemNamed: "leo.close") {
+      let button = UIButton()
+      button.setImage(buttonImage, for: .normal)
+      button.imageView?.contentMode = .scaleAspectFit
+      button.imageView?.tintColor = .white
+      button.imageView?.preferredSymbolConfiguration = .init(
+        font: .preferredFont(for: .title3, weight: .regular),
+        scale: .small
+      )
+
+      button.imageView?.snp.makeConstraints {
+        $0.width.equalTo(SearchResultAdClickedInfoBarUX.toastCloseButtonWidth)
+      }
+
+      button.addGestureRecognizer(
+        UITapGestureRecognizer(target: self, action: #selector(buttonPressed))
+      )
```

Comment:
No need to use a gesture recognizer to a `UIButton`, just use `addTarget` instead

Reply:
Fixed